src/ddmd/delegatize.d

```diff
@@ -170,12 +170,23 @@ extern (C++) bool lambdaCheckForNestedRef(Expression e, Scope* sc)
     return v.result;
 }
 
-bool checkNestedRef(Dsymbol s, Dsymbol p)
+/*****************************************
+ * See if context `s` is nested within context `p`, meaning
+ * it `p` is reachable at runtime by walking the static links.
+ * If any of the intervening contexts are function literals,
+ * make sure they are delegates.
+ * Params:
+ *      s = inner context
+ *      p = outer context
+ * Returns:
+ *      true means it is accessible by walking the context pointers at runtime
+ */
+bool ensureStaticLinkTo(Dsymbol s, Dsymbol p)
```

Comment:
We should use the accepted compiler jargon term rather than invent our own. I'll add the link you found to the function comment, though.